internal/formatter/formatter.go

```diff
@@ -0,0 +1,58 @@
+package formatter
+
+import (
+	"sort"
+	"sync"
+
+	"github.com/docker/app/internal/formatter/driver"
+	composetypes "github.com/docker/cli/cli/compose/types"
+	"github.com/pkg/errors"
+)
+
+var (
+	driversMu sync.RWMutex
+	drivers   = map[string]driver.Driver{}
+)
+
+// Register makes a formatter available by the provided name.
+// If Register is called twice with the same name or if driver is nil,
+// it panics.
+func Register(name string, driver driver.Driver) {
+	driversMu.Lock()
+	defer driversMu.Unlock()
+	if driver == nil {
```

Comment:
Can drive be `nil` as it's not exported and cannot be manipulated outside of this package?

Reply:
@silvin-lubecki `driver` is a `driver.Driver` passed as argument. As it's on `internal`, it's kinda safe to assume it won't be nil but better safe than sorry (especially if we move that package out of `internal` some day)

Reply:
Oh sorry misread (`drivers`), forget my comment!